pkg/util/wimi/iswimi.go

```diff
@@ -0,0 +1,14 @@
+package wimi
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "github.com/Azure/ARO-RP/pkg/api"
+
+// IsWimi checks whether a cluster is a Workload Identity cluster or Service Principal cluster
+func IsWimi(oc *api.OpenShiftCluster) bool {
+	if oc.Properties.PlatformWorkloadIdentityProfile == nil || oc.Properties.ServicePrincipalProfile != nil {
+		return false
+	}
+	return true
+}
```

Comment:
```suggestion
	if oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil {
		return true
	}
	return false
}
```

If we get invalid cluster (e.g.  PlatformWorkloadIdentityProfile != nil and ServicePrincipalProfile != nil), it returns true, which is not correct.

We might not have to care about invalid settings, but it would be more intuitive that only when it has valid settings and WI is not nil, it returns true.

Reply:
done, thank you!